main.go

```diff
@@ -46,49 +47,28 @@ func main() {
 		Key:     string(key),
 		BaseURL: chefServerUrl,
 	})
-	if err != nil {
-		fmt.Println("Unable setup Chef client:", err)
-		os.Exit(1)
-	}
 
-	// List Nodes
-	nodesList, err := client.Nodes.List()
 	if err != nil {
-		fmt.Println("Unable to list nodes:", err)
+		fmt.Println("Unable setup Chef client:", err)
 		os.Exit(1)
 	}
 
-	// Print out the list
-	fmt.Println("Nodes:")
-	for nodeName, _ := range nodesList {
-		fmt.Println(nodeName)
-	}
+	query := map[string]interface{}{
+		"name":         []string{"name"},
+		"chef_version": []string{"chef_packages", "chef", "version"},
+		"os":           []string{"platform"},
+		"os_version":   []string{"platform_version"}}
 
-	// List Cookbooks
-	cookbookList, err := client.Cookbooks.List()
+	pres, err := client.Search.PartialExec("node", "*:*", query)
 	if err != nil {
-		fmt.Println("Unable to list cookbooks:", err)
-		os.Exit(1)
-	}
-
-	// Print out the list
-	fmt.Println("\nCookbooks:")
-	for cookbookName, cookbookVersions := range cookbookList {
-		versionsArray := cookbookVersionsAsArrayStrings(&cookbookVersions)
-		fmt.Printf("%s %v\n", cookbookName, versionsArray)
+		log.Fatal("Error running Search.PartialExec()", err)
 	}
-}
-
-// TODO @afiune contribute back to the community repo, propose to add a function to the struct
```

Comment:
I still think this is a good use case to contribute, though I will add this to my personal TODO 😉